### PR TITLE
NOISSUE - Fix Update user roles, Listing and retrieve users with role 

### PR DESCRIPTION
**Pull request by arvindh123**
Pull request title should be `MG-XXX - description` or `NOISSUE - description` where XXX is ID of issue that this PR relate to.
Please review the [CONTRIBUTING.md](https://github.com/absmach/magistrala/blob/master/CONTRIBUTING.md) file for detailed contributing guidelines.

### What does this do?
- Fix : Update user role and  Listing and retrieve users with role 

### Update user role to admin 
Request: 
```bash 
curl --location --request PATCH 'http://localhost/users/28cea756-f23b-408d-ab7a-43bf6f3c20cd/role' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'Authorization: Bearer <TOKEN>' \
--data '{
  "role": "admin"
}'
```
Response: 
```json 
{
    "id": "28cea756-f23b-408d-ab7a-43bf6f3c20cd",
    "credentials": {
        "identity": "user1@example.com"
    },
    "created_at": "2023-11-23T09:03:11.411868Z",
    "updated_at": "2023-11-23T16:35:17.597552Z",
    "updated_by": "8cadfa56-e513-4dd8-b746-073348432054",
    "role": "admin"
}
```
### Listing of users 
Request : 
```bash 
curl --location 'http://localhost/users' \
--header 'Accept: application/json' \
--header 'Authorization: Bearer <TOKEN>'
```

Response : 
```json
{
    "limit": 10,
    "total": 3,
    "users": [
        {
            "id": "8cadfa56-e513-4dd8-b746-073348432054",
            "name": "admin",
            "credentials": {
                "identity": "admin@example.com"
            },
            "metadata": {
                "role": "admin"
            },
            "created_at": "2023-11-23T08:56:44.364784Z",
            "updated_at": "0001-01-01T00:00:00Z",
            "role": "admin"
        },
        {
            "id": "28cea756-f23b-408d-ab7a-43bf6f3c20cd",
            "credentials": {
                "identity": "user1@example.com"
            },
            "created_at": "2023-11-23T09:03:11.411868Z",
            "updated_at": "2023-11-23T16:35:17.597552Z",
            "updated_by": "8cadfa56-e513-4dd8-b746-073348432054",
            "role": "admin"
        },
        {
            "id": "aaef1e86-3968-4336-89e9-5b2d01bc416c",
            "credentials": {
                "identity": "user2@example.com"
            },
            "created_at": "2023-11-23T09:03:52.378677Z",
            "updated_at": "0001-01-01T00:00:00Z"
        }
    ]
}
```
### Which issue(s) does this PR fix/relate to?
Put here `Resolves #XXX` to auto-close the issue that your PR fixes (if such)

### CODE REVIEW

File: api/openapi/users.yml [hunk 1]
```diff
@@ -1658,13 +1656,13 @@ components:
           schema:
             $ref: "#/components/schemas/UserSecret"
 
-    UserUpdateOwnerReq:
+    UserUpdateRoleReq:
       description: JSON-formated document describing the owner of user to be update
```
Comment (ianmuchyri) [inline api/openapi/users.yml:1660]:
```suggestion
      description: JSON-formated document describing the role of the user to be update
```

**Review by ianmuchyri — CHANGES_REQUESTED** [review]

**Review by ianmuchyri — CHANGES_REQUESTED** [review]
In sdk.go change `UpdateUserOwner` to `UpdateUserRole`

**Comment by ianmuchyri** [comment]
> In sdk.go change `UpdateUserOwner` to `UpdateUserRole`

also update in `users_tests.go`

**Comment by arvindh123** [comment]
> UpdateUserOwner

Fixed in https://github.com/absmach/magistrala/pull/49

File: api/openapi/users.yml [hunk 2]
```diff
@@ -1658,13 +1656,13 @@ components:
           schema:
             $ref: "#/components/schemas/UserSecret"
 
-    UserUpdateOwnerReq:
-      description: JSON-formated document describing the owner of user to be update
+    UserUpdateRoleReq:
+      description: JSON-formated document describing the role of the user to be update
```
[the remark this answers is not in the document]
Comment (dborovcanin) [reply api/openapi/users.yml:1660]: @arvindh123 Please address this one.

**Review by dborovcanin — APPROVED** [review]

**Pull request merged**